core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -49,6 +52,17 @@ public static ObjectMapper mapper() {
     return MAPPER;
   }
 
+  public static ObjectMapper withRESTComponents() {
+    RESTSerializers.registerAll(MAPPER);
+    // These are a workaround for Jackson since Iceberg doesn't use the standard get/set bean notation.
+    // This allows Jackson to work with the fields directly (both public and private) and not require
+    // custom (de)serializers for all the request/response objects.
+    MAPPER.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE);
+    MAPPER.setVisibility(PropertyAccessor.CREATOR, JsonAutoDetect.Visibility.ANY);  // deserialization
+    MAPPER.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);  // serialization
+    return MAPPER;
+  }
+
```

Comment:
These changes are part of the other PR and so should be reviewed there. They're included here for test purposes.

Reply:
I thought we talked about removing these. Can't the tests instantiate their own mapper?

Reply:
I didn't think we really came to a conclusion about removing these. I know we talked about removing annotations (`@JsonProperty`).

The tests can definitely instantiate their own mappers, but without these somewhere we'll need to write out the Serializer / Deserializer for all of these (or make the fields public - which even then some mappers might be able to turn these off).

I'll move these to the test classes for now though.

Reply:
Dealt with this in the other PR. Will be removing this from this PR entirely.